library/core/src/panicking.rs

```diff
@@ -92,6 +93,20 @@ pub fn panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
     unsafe { panic_impl(&pi) }
 }
 
+/// This function is used instead of panic_fmt in const eval.
+#[cfg(not(bootstrap))]
+#[lang = "const_panic_fmt"]
+pub const fn const_panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
+    if let Some(msg) = fmt.as_str() {
+        panic_str(msg);
+    } else {
+        // SAFETY: This is only evaluated at compile time, which handles this
+        // fine (in case it turns out this branch turns out to be reachable
```

Comment:
```suggestion
        // SAFETY: This is only evaluated at compile time, which reliably
        // handles this UB (in case this branch turns out to be reachable
```